spec/pagseguro/authorization_request/request_serializer_spec.rb

```diff
@@ -1,23 +1,285 @@
   end
-  let(:authorization) { PagSeguro::AuthorizationRequest.new(options) }
-  let(:serializer) { described_class.new(authorization) }
-  let(:params) { serializer.to_params }
 
-  it{ expect(params).to include(credentials: credentials) }
-  it{ expect(params).to include(permissions: 'CREATE_CHECKOUTS,RECEIVE_TRANSACTION_NOTIFICATIONS') }
-  it{ expect(params).to include(notificationURL: 'foo.com') }
-  it{ expect(params).to include(redirectURL: 'bar.com') }
-  it{ expect(params).to include(reference: 'ref1234') }
+  let(:authorization_request) { PagSeguro::AuthorizationRequest.new(options) }
+  let(:serializer) { described_class.new(authorization_request) }
+  let(:xml) { serializer.build_xml }
+  let(:xml_parsed) { Nokogiri::XML(xml) }
+  let(:node) { xml_parsed.at_css('authorizationRequest') }
+
+  it 'encoding should be equal to application encoding' do
+    encoding = xml_parsed.encoding
+    expect(encoding).to eq PagSeguro.encoding
+  end
+
+  it 'must have a authorizationRequest' do
+    expect(node).to be_a_instance_of Nokogiri::XML::Element
+  end
+
+  it 'must have the correct permissions' do
+    permissions = options[:permissions].map{|value| PagSeguro::AuthorizationRequest::PERMISSIONS[value] }
+    permissions_code_parsed = xml_parsed.css('authorizationRequest > permissions > code').map(&:text)
+
+    expect(permissions).to eq permissions_code_parsed
+  end
+
+  it 'must have a redirect URL' do
+    redirect_url_parsed = xml_parsed.at_css('authorizationRequest > redirectURL').text
+    expect(redirect_url_parsed).to eq options[:redirect_url]
+  end
+
+  it 'must have a notification URL' do
+    notification_url = xml_parsed.at_css('authorizationRequest > notificationURL').text
+    expect(notification_url).to eq options[:notification_url]
+  end
+
+  context 'when an account is assigned' do
+    let(:account) { {} }
+    let(:options) { super().merge(account: account) }
+    let(:node) { super().at_css('> account') }
+
+    it 'can be an email' do
+      account[:email] = 'usuario@seusite.com.br'
+      expect( node.at_css('> email').text ).to eq account[:email]
+    end
+
+    it 'can be a type' do
+      account[:type] = 'SELLER'
+      expect( node.at_css('> type').text ).to eq account[:type]
+    end
+
+    context 'when a person is assigned'  do
+      let(:person) { {} }
+      let(:account) { super().merge(person: person) }
+      let(:node) { super().at_css('> person') }
+
+      it 'can be name' do
+        person[:name] = 'Antonio Carlos'
+        expect( node.at_css('> name').text ).to eq person[:name]
+      end
+
+      it 'can be birth date' do
+        person[:birth_date] = '1982-02-05'
+        expect( node.at_css('> birthDate').text ).to eq person[:birth_date]
+      end
+
+      context 'when the address is assigned' do
+        let(:address) { {} }
+        let(:person) { super().merge(address: address) }
+        let(:node) { super().at_css('> address') }
+
+        include_examples 'when address is assigned'
```

Comment:
yes, is almost the same thing, but `include_examples` show:
```
when the address is assigned
        can be a postal code
        can be a street
        can be a number
        can be a complement
        can be a district
```

and the `it_behaves_like`:

```
SomeClass
  behaves like SharedExample
    should be initialize with type equal ...
    should not be possible to ...
    should accepts_nested_attributes_for ...
```